### PR TITLE
MAP-1287 /select-residential-location to gets groups from locationsApi

### PR DESCRIPTION
**Pull request by GurnankCheema**
see Jira [MAP-1287](https://dsdmoj.atlassian.net/jira/software/c/projects/MAP/boards/1354?quickFilter=3684&selectedIssue=MAP-1287)

Replacing call to whereaboutsApi.searchGroups() with locationsInsidePrisonApi.getSearchGroups() for the selectResidentialLocation.ts module.

There are other module's that still rely on whereaboutsApi.searchGroups() so it has not been removed from the codebase yet. 

[MAP-1287]: https://dsdmoj.atlassian.net/browse/MAP-1287?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: backend/api/locationsInsidePrisonApi.test.ts [hunk 1]
```diff
@@ -60,4 +60,62 @@ describe('Locations Inside Prison Api', () => {
       ).rejects.toThrow('Bad Request')
     })
   })
+
```
Comment (GurnankCheema) [inline backend/api/locationsInsidePrisonApi.test.ts:63]: Lines 64 to 92 relate to a test missed during previous code changes so added now

File: backend/api/locationsInsidePrisonApi.ts [hunk 2]
```diff
@@ -25,12 +34,17 @@ export const locationsInsidePrisonApiFactory = (client: OauthApiClient) => {
   const getAgencyGroupLocationPrefix = (systemContext, prisonId, group): Promise<LocationPrefix> =>
     getWith404AsNull(systemContext, `/locations/prison/${prisonId}/group/${group}/location-prefix`)
 
-  const getAgencyGroupLocations = (systemContext, prisonId, groupName): Promise<Location> =>
+  const getAgencyGroupLocations = (systemContext, prisonId, groupName): Promise<Location[]> =>
     getWith404AsNull(systemContext, `/locations/groups/${prisonId}/${groupName}`)
```
Comment (GurnankCheema) [inline backend/api/locationsInsidePrisonApi.ts:39]: getSearchGroups calls the locationsApi

File: backend/controllers/selectResidentialLocation.ts [hunk 3]
```diff
@@ -1,15 +1,21 @@
...
     const today = moment()
     const { errors, formValues } = pageData || {}
     try {
       const {
         user: { activeCaseLoad },
       } = res.locals
-      const residentialLocations = await whereaboutsApi.searchGroups(res.locals, activeCaseLoad.caseLoadId)
+
+      const systemContext = await systemOauthClient.getClientCredentialsTokens(req.session.userDetails.username)
+
```
Comment (GurnankCheema) [inline backend/controllers/selectResidentialLocation.ts:14]: The key change in this PR

File: cypress.config.js [hunk 4]
```diff
@@ -468,6 +470,9 @@ module.exports = defineConfig({
         verifyMoveToCell: (body) => prisonApi.verifyMoveToCell(body),
         stubGetLocationPrefix: ({ agencyId, groupName, response }) =>
           locationsInsidePrisonApi.stubGetLocationPrefix({ agencyId, groupName, response }),
```
Comment (GurnankCheema) [inline cypress.config.js:472]: adding stubGetGroupSearch and then grouping the stubs for locationsInsidePrisonApi together

File: backend/controllers/selectResidentialLocation.ts (hunk 3 above)
Comment (marcusaleman) [reply backend/controllers/selectResidentialLocation.ts:14]: how the secrests are managed in this project?

**Review by marcusaleman — COMMENTED** [review]
+ LGTM just i have had some checking.
+ I have a query about the 'secrets'

**Review by Mjwillis — APPROVED** [review]
LGTM

**Pull request merged**